#### 使用finally在promise用法

finally永远执行

```js
class ParamError extends Error {
    constructor(msg) {
        super(msg);
        this.name = "ParamError";
    }
}

class HttpError extends Error {
    constructor(msg) {
        super(msg);
        this.name = "Http Error"
    }
}

function ajax(url) {
    isShow = true;
    return new Promise((resolve, reject) => {
        if (!/^http/.test(url)) {
            throw new ParamError("请求地址错误")
        }
        let xhr = new XMLHttpRequest();
        xhr.open("GET", url);
        xhr.send();
        xhr.onload = function () {
            if (this.status == 200) {
                resolve(JSON.parse(this.response))
            } else if (this.status == 404) {
                reject(new HttpError("网络请求异常"))
            } else {
                reject("加载失败")
            }
        }
        xhr.onerror = function () {
            reject(this);
        }
    })
}
```



```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <style>
        div{
            width: 100px;
            height: 100px;
            background: red;
            color: white;
        }
    </style>
</head>

<body>
    
    <div>loading</div>
</body>
<script>
    const promise = new Promise((resolve,reject) => {
        reject("hdcms")
    }).then(msg => {
        console.log("resolve")
    }).catch(err => {
        console.log(err)
    }).finally(() => {
        isShow = false;
        console.log("finally")
    })
</script>
</html>
```

